### PR TITLE
chore: update go-github to V65

### LINKED ISSUE #2358: [MAINT]: Update version of go-github
### Describe the need

New features have been added to go-github and released in V65, I will raise a PR to implement this.

### SDK Version

go-github v65

### API Version

_No response_

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by grahamhar**
Resolves #2358 

----

### Before the change?

* go-github@v63 

### After the change?

* go-github@v65

### Pull request checklist
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)

### Does this introduce a breaking change?

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/master/community/breaking_changes.md) to help!

- [ ] Yes
- [ ] No

----

I have attempted to run the checklist but some changes are being flagged in plan due to dependabot being enabled by default on the org in use. Is this due to the way new orgs have vulnerability_alerts on by default now for public repositories? 

Example of failing test.

```sh
RUN   TestAccGithubRepositoryFileDataSource/create_and_read_a_file_without_providing_a_branch_name/with_an_organization_account
    data_source_github_repository_file_test.go:166: Step 1/1 error: After applying this test step, the plan was not empty.
        stdout:
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place
         <= read (data resources)
        
        Terraform will perform the following actions:
        
          # data.github_repository_file.test will be read during apply
          # (depends on a resource or a module with changes pending)
         <= data "github_repository_file" "test" {
              + commit_author  = (known after apply)
              + commit_email   = (known after apply)
              + commit_message = (known after apply)
              + commit_sha     = (known after apply)
              + content        = (known after apply)
              + file           = "test"
              + id             = (known after apply)
              + ref            = (known after apply)
              + repository     = "tf-acc-test-2ly8t"
              + sha            = (known after apply)
            }
        
          # github_repository.test will be updated in-place
          ~ resource "github_repository" "test" {
                id                          = "tf-acc-test-2ly8t"
                name                        = "tf-acc-test-2ly8t"
              - vulnerability_alerts        = true -> null
                # (32 unchanged attributes hidden)
        
                # (1 unchanged block hidden)
            }
        
        Plan: 0 to add, 1 to change, 0 to destroy.
--- FAIL: TestAccGithubRepositoryFileDataSource (28.72s)
    --- FAIL: TestAccGithubRepositoryFileDataSource/create_and_read_a_file_with_a_branch_name_provided (11.38s)
        --- SKIP: TestAccGithubRepositoryFileDataSource/create_and_read_a_file_with_a_branch_name_provided/with_an_anonymous_account (0.00s)
        --- SKIP: TestAccGithubRepositoryFileDataSource/create_and_read_a_file_with_a_branch_name_provided/with_an_individual_account (0.00s)
        --- FAIL: TestAccGithubRepositoryFileDataSource/create_and_read_a_file_with_a_branch_name_provided/with_an_organization_account (11.38s)
    --- FAIL: TestAccGithubRepositoryFileDataSource/create_and_read_a_file_without_providing_a_branch_name (17.35s)
        --- SKIP: TestAccGithubRepositoryFileDataSource/create_and_read_a_file_without_providing_a_branch_name/with_an_anonymous_account (0.00s)
        --- SKIP: TestAccGithubRepositoryFileDataSource/create_and_read_a_file_without_providing_a_branch_name/with_an_individual_account (0.00s)
        --- FAIL: TestAccGithubRepositoryFileDataSource/create_and_read_a_file_without_providing_a_branch_name/with_an_organization_account (17.34s)
FAIL
FAIL    github.com/integrations/terraform-provider-github/v6/github     29.802s
FAIL

```

### CODE REVIEW

File: github/data_source_github_organization_team_sync_groups.go
Comment (zepeng811) [inline github/data_source_github_organization_team_sync_groups.go:45]: This is removing the `PerPage` option which is changing the behaviour

perhaps this should be
```suggestion
	options := &github.ListIDPGroupsOptions{
	    ListCursorOptions: github.ListCursorOptions{
	        PerPage: maxPerPage,
        },
	}
```

File: go.mod
Comment (zepeng811) [inline go.mod:11]: we need to bump this to v65 and remove the indirect below, running `go mod tidy` should then also remove v64 from `go.sum`

Comment (MXfive) [reply go.mod:11]: https://github.com/integrations/terraform-provider-github/pull/2380/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6

Comment (grahamhar) [reply go.mod]: Sorry I was rushing and missed this step, done now

Comment (zepeng811) [reply go.mod]: thanks!

**Review by zepeng811 — APPROVED** [review]

**Comment by zepeng811** [comment]
@kfcampbell @nickfloyd (sorry just tagging some maintainers who recently approved PRs)

can we get some eyes on this PR please. It helps unblock couple other PRs that is currently blocked requiring newer version of google/go-github

**Comment by MXfive** [comment]
@grahamhar @kfcampbell @nickfloyd 

This is has been open for over a month and blocking a few other PRs now, possible to have this merged or taken over by someone with bandwidth to do so?

**Comment by nickfloyd** [comment]
> @grahamhar @kfcampbell @nickfloyd
> 
> This is has been open for over a month and blocking a few other PRs now, possible to have this merged or taken over by someone with bandwidth to do so?

Apologies for the delay.  I'm looking at this today and should have it merged in and shipped.

**Review by kfcampbell — APPROVED** [review]
> I have attempted to run the checklist but some changes are being flagged in plan due to dependabot being enabled by default on the org in use. Is this due to the way new orgs have vulnerability_alerts on by default now for public repositories?

> Example of failing test.

@grahamhar unfortunately this failure has crept into our main branch and will need to be dealt with separately. Fortunately for this case, this means we can take this PR as-is now. 

As always, thank you for your attention to the testing in this provider.

**Pull request merged**
